src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/utils/TaskMandatoryFieldsValidator.java

```diff
@@ -0,0 +1,139 @@
+
+            if (mandatoryFieldCheckEnabledServices == null) {
+                log.warn("Mandatory field check enabled services is null, skipping validation.");
+                return;
+            }
+
+            JsonNode excludedJurisdictionsArray = parseJson(mandatoryFieldCheckEnabledServices.toJsonString());
+            if (excludedJurisdictionsArray == null) {
+                log.warn("Excluded jurisdictions array is null, skipping jurisdiction check.");
+                validateTaskMandatoryFields(task);
+                return;
+            }
+
+            if (!isJurisdictionExcluded(task, excludedJurisdictionsArray)) {
+                validateTaskMandatoryFields(task);
+            }
+        }
+    }
+
+    /**
+     * Checks if the task's jurisdiction is excluded.
+     *
+     * @param task the task to be checked
+     * @param excludedJurisdictionsArray the array of excluded jurisdictions
+     * @return true if the jurisdiction is excluded, false otherwise
+     */
+    private boolean isJurisdictionExcluded(TaskResource task, JsonNode excludedJurisdictionsArray) {
+        for (JsonNode jurisdiction : excludedJurisdictionsArray) {
+            if (jurisdiction.asText().equals(task.getJurisdiction())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Parses a JSON string and returns the JsonNode for the mandatory field check flag variant.
+     *
+     * @param jsonString the JSON string to be parsed
+     * @return the JsonNode for the mandatory field check flag variant
+     * @throws IllegalArgumentException if the JSON string cannot be parsed
+     */
+    public JsonNode parseJson(String jsonString) {
+        try {
+            return new ObjectMapper().readTree(jsonString).get(MANDATORY_FIELD_CHECK_FLAG_VARIANT.stringValue());
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("Mandatory flag jurisdictions mapping issue.", e);
+        }
+    }
+
+    /**
+     * Validates the mandatory fields of a task.
+     *
+     * @param task the task to be validated
+     * @throws ValidationException if any mandatory field is missing or invalid
+     * @throws IllegalArgumentException if a property value cannot be found
+     */
+    public void validateTaskMandatoryFields(TaskResource task) {
+        List<String> errors = new ArrayList<>();
+        for (String field : taskMandatoryFields) {
+            try {
+                if (PropertyUtils.getProperty(task, field) == null
+                    || PropertyUtils.getProperty(task, field).toString().isBlank()) {
+                    errors.add(field + " cannot be null or empty");
+                }
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Cannot find property value for field " + field, e);
+            }
+        }
+        if (!errors.isEmpty()) {
+            throw new ValidationException(String.join(", ", errors));
```

Comment:
Something to consider here is that there are 2 flavours of mandatory field violations.  1) service team has not provided one though a config error 2) we have not provided one through our code.  The former is going to be what we are most interested in since with the latter some test is likely to fail.  Where the services have made a mistake we want to make sure we alert then asap and so we have to know it was a field which was their responsibility that was the problem (not our code).
Given this our model should reflect this distinction.  One can be used to alert the service but the other not.